fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt

```diff
@@ -2247,20 +2352,34 @@ class FIOSDK(private var privateKey: String, var publicKey: String,var technolog
     }
 
     private fun validateAddPublicAddress(fioAddress:String, chainCode:String, tokenCode:String,
-                                         tokenPublicAddress:String,
-                                         technologyPartnerId:String=""): Validator
+                                           tokenPublicAddress:String,
+                                           technologyPartnerId:String=""): Validator
+{
+    var isValid = (fioAddress.isFioAddress()
+            && tokenPublicAddress.isNativeBlockChainPublicAddress()
+            && tokenCode.isTokenCode() && chainCode.isTokenCode())
+
+    if(technologyPartnerId.isNotEmpty())
+        isValid = isValid && technologyPartnerId.isFioAddress()
+
+    return Validator(isValid,if(!isValid) "Invalid AddPublicAddress Request" else "")
+}
+
+    private fun validateAddPublicAddresses(fioAddress:String,
+                                           tokenPublicAddresses:List<TokenPublicAddress>,
+                                           technologyPartnerId:String=""): Validator
     {
-        var isValid = (fioAddress.isFioAddress()
-                && tokenPublicAddress.isNativeBlockChainPublicAddress()
-                && tokenCode.isTokenCode() && chainCode.isTokenCode())
+        var isValid = true
 
-        if(technologyPartnerId.isNotEmpty())
-            isValid = isValid && technologyPartnerId.isFioAddress()
+        for(address in tokenPublicAddresses)
+        {
+            isValid = isValid && this.validateAddPublicAddress(fioAddress,address.chainCode,address.tokenCode,address.publicAddress,technologyPartnerId).isValid
+        }
 
         return Validator(isValid,if(!isValid) "Invalid AddPublicAddress Request" else "")
```

Comment:
Should say "Invalid AddPublicAddresses Request"

Reply:
refactored, so names make more sense